### ISSUE TITLE
uninitialized constant GraphQL::Language::Nodes::NONE (NameError)

### ISSUE THREAD

**Issue by paneq** [rmosolgo/graphql-ruby#4607]
**Describe the bug**

adding `graphql-c_parser` to `Gemfile` leads to `uninitialized constant GraphQL::Language::Nodes::NONE (NameError)` exception when trying to boot an application.

**Versions**

`graphql` version: 2.0.11
`rails`: 7.04
`graphql-c_parser`: 1.0.5 or 1.0.4

**Steps to reproduce**

Use the combination of above gems and run `bundle exec rails c`

**Expected behavior**

I expected rails console to boot.

**Actual behavior**

Exception `uninitialized constant GraphQL::Language::Nodes::NONE`

Place full backtrace here (if a Ruby exception is involved):

<details>
<summary>Click to view exception backtrace</summary>

```
<internal:/home/rupert/.rbenv/versions/3.2.1/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require': uninitialized constant GraphQL::Language::Nodes::NONE (NameError)
	from <internal:/home/rupert/.rbenv/versions/3.2.1/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from /home/rupert/.rbenv/versions/3.2.1/lib/ruby/gems/3.2.0/gems/bootsnap-1.11.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
	from /home/rupert/.rbenv/versions/3.2.1/lib/ruby/gems/3.2.0/gems/zeitwerk-2.6.11/lib/zeitwerk/kernel.rb:38:in `require'
	from /home/rupert/.rbenv/versions/3.2.1/lib/ruby/gems/3.2.0/gems/graphql-c_parser-1.0.4/lib/graphql/c_parser.rb:5:in `<main>'
	from <internal:/home/rupert/.rbenv/versions/3.2.1/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from <internal:/home/rupert/.rbenv/versions/3.2.1/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from /home/rupert/.rbenv/versions/3.2.1/lib/ruby/gems/3.2.0/gems/bootsnap-1.11.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
	from /home/rupert/.rbenv/versions/3.2.1/lib/ruby/gems/3.2.0/gems/zeitwerk-2.6.11/lib/zeitwerk/kernel.rb:38:in `require'
	from /home/rupert/.rbenv/versions/3.2.1/lib/ruby/gems/3.2.0/gems/graphql-c_parser-1.0.4/lib/graphql-c_parser.rb:2:in `<main>'
	from <internal:/home/rupert/.rbenv/versions/3.2.1/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from <internal:/home/rupert/.rbenv/versions/3.2.1/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:37:in `require'
	from /home/rupert/.rbenv/versions/3.2.1/lib/ruby/gems/3.2.0/gems/bootsnap-1.11.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:30:in `require'
	from /home/rupert/.rbenv/versions/3.2.1/lib/ruby/gems/3.2.0/gems/zeitwerk-2.6.11/lib/zeitwerk/kernel.rb:38:in `require'
	from /home/rupert/.rbenv/versions/3.2.1/lib/ruby/gems/3.2.0/gems/bundler-2.4.13/lib/bundler/runtime.rb:60:in `block (2 levels) in require'
	from /home/rupert/.rbenv/versions/3.2.1/lib/ruby/gems/3.2.0/gems/bundler-2.4.13/lib/bundler/runtime.rb:55:in `each'
	from /home/rupert/.rbenv/versions/3.2.1/lib/ruby/gems/3.2.0/gems/bundler-2.4.13/lib/bundler/runtime.rb:55:in `block in require'
	from /home/rupert/.rbenv/versions/3.2.1/lib/ruby/gems/3.2.0/gems/bundler-2.4.13/lib/bundler/runtime.rb:44:in `each'
	from /home/rupert/.rbenv/versions/3.2.1/lib/ruby/gems/3.2.0/gems/bundler-2.4.13/lib/bundler/runtime.rb:44:in `require'
	from /home/rupert/.rbenv/versions/3.2.1/lib/ruby/gems/3.2.0/gems/bundler-2.4.13/lib/bundler.rb:187:in `require'
	from /home/rupert/development/toptal/platform/config/application.rb:14:in `<main>'

```

</details>

**Comment by rmosolgo** [comment]
Oops -- sorry for the trouble! It looks like graphql/c_parser depends on this commit: 

Which was released in GraphQL-Ruby 2.0.20. I'll update the gemspec for graphql-c_parser soon; in the meantime you can either update to GraphQL-Ruby 2.0.20 or higher, or add the constant yourself: 

- Tell bundler not to auto-load the gem: 

    ```ruby
    gem "graphql/c_parser", require: false # it causes constant not found error without our shim 
    ```
    
- Define the constant in your schema file, then load the gem manually: 

  ```ruby 
  # app/graphql/my_schema.rb 
  # This constant is required by graphql/c_parser but only available in GraphQL-Ruby 2.0.20+
  GraphQL::Language::Nodes::NONE = [].freeze 
  require "graphql/c_parser"
  ```
  
Does one of those approaches work for you for now?

**Comment by paneq** [comment]
Hey, thank you for extremely fast response. Much appreciated. Since I have multiple schemas in my project, I went with something inspired by your second proposal.

```ruby
# Gemfile
gem 'graphql', '= 2.0.11'
gem 'graphql-c_parser', '~> 1.0', '= 1.0.4', require: false
```

and

```ruby
# config/initializers/graphql_parser.rb

# https://github.com/rmosolgo/graphql-ruby/issues/4607
# This constant is required by graphql/c_parser but only available in GraphQL-Ruby 2.0.20+
if Gem::Version.new(GraphQL::VERSION) < Gem::Version.new("2.0.20")
  GraphQL::Language::Nodes::NONE = [].freeze
  require "graphql/c_parser"
else
  raise "Remove this file and remove 'require: false' from Gemfile entry for gem 'graphql-c_parser'"
end
```

**Comment by rmosolgo** [comment]
I just released graphql-c_parser v1.0.6 which doesn't use this constant anymore. Could you give that version a try? 

I'm not sure about compatibility 1.13.x; it might work just fine because I think the AST code has been pretty stable since then, but I haven't tried it!
